### DISCUSSION TITLE
Generate Pole Figure from Multiple Measurements

### DISCUSSION THREAD

**Discussion by sarialzerey** [mtex-toolbox/mtex#2044] [Ask Anything] [answered]
Hi,

If I have more than one measurement and I want to compile the data together in order to generate a pole figure or grain size distribution. How do I do it? 

It’s important to mention that after I process the first measurement and let’s say delete the small grains (<5 pixels) so that the generated pole figure from the multiple measurements reflect all my changes. 

In general, every time I need an information, I reprocess the data from scratch because I don’t know how to store the processed data. 

Finally, I still don’t understand how the variables are stored inside EBSD variable. How can I understand it?

thanks!!

**Comment by kilir** [comment]
Hi,
in general you can concatenate most data and save whatever in Matlab as a .mat.

very generic example:

```
% load data, compute grains and save as individual .mat
inputfiles = dir('*.ang')

for i = 1:length(inputfiles)
ebsd = EBSD.load(inputfiles(i).name)
grains = ebsd.calcGrains

save([inputfiles(i).name(1:end-4) 'EbsdGrainsxyzblabla.mat'], 'grains', 'ebsd')
end


% load  .mat and e.g. gather all mean orientations 
inputfiles = dir('*EbsdGrainsxyzblabla.mat')
meanO=[]
for i = 1:inputfiles
load(inputfiles(i).name)
meanO = [meanO; grains('phasename').meanOrientation]
end

```
>Finally, I still don’t understand how the variables are stored inside EBSD variable. How can I understand it?

It's just a structure with properties, fields containing all the data. Have a look at the contructor @EBSD/EBSD.m

Hope that helps.

Cheers, Rüdiger

**Reply by sarialzerey** [reply]
Thanks for the response. However, yes it compiled the files but when I run the code I get the IPF only for one measurement (last one)

I also suspect that I'm not selecting the right IPFcolorkey. Can you please check that?
I simply want to generate IPF along x, y and z direction

%% IPF
figure();clc;
gB_Cu = grains.boundary('copper','copper')
% computed the axes in specien coordinates
axes = gB_Cu.misorientation.axis
% define a color key
colorKey = HSVDirectionKey(axes);
% compute colors
color = colorKey.direction2color(axes);
plot(colorKey); hold on;
plotIPDF(ebsd('copper').orientations,zvector,'MarkerColor','black','MarkerSize',1)



%% IPF distribution color map
figure();clc;
% define a color key
colorKey = HSVDirectionKey(axes);
% compute colors
color = colorKey.direction2color(axes);
plotIPDF(ebsd('copper').orientations,[vector3d.X,vector3d.Y,vector3d.Z],'contourf','smooth')
mtexColorbar

**Comment by sarialzerey** [comment]
Can you guys help me with this please

**Reply by zmichels** [reply]
Which part are you needing help with exactly?

The process (as Rudiger explained) requires concatenating/combining data from saved files/datasets. The tricky part in doing that is to isolate (and maybe rename) the variables you want to combine before loading the next dataset -- thereby avoiding a case in which the newly loaded dataset overwrites the variables of interest from the first dataset. And, when isolating the data from each subsequently loaded dataset... you want to concatenate those data into the variable/s you are keeping between each dataset load.

Does that make sense? I'd say it's more fiddly than "difficult", but it requires some data juggling. The good news is that once you get the data combined into your workspace, you can save that combined data as a .MAT file to open later -- so you don't have to undertake the steps of combining each time. Further, if it's the type of task you will need to do regularly with other groups of data... you could write a script to perform the task on [for instance] all the MAT files in a specific folder or something.

**Reply by zmichels** [reply, marked as answer]
Some thoughts that might help...

When you load a previously processed dataset, you could load only the variables you are interested in (and avoid conflicts with others)... see the explanation from Mathworks at this link:
https://www.mathworks.com/help/matlab/ref/load.html#d126e920093

When clearing variables in the workspace (like maybe before loading in another file)... you can use an "except" flag to clear all the variables except certain ones (leaving them in the workspace to concatenate with subsequently loaded data). See the link here regarding the "clearvars -except" syntax:
https://www.mathworks.com/help/matlab/ref/clearvars.html#d126e198525

**Reply by sarialzerey** [reply]
Thanks so much! 
In addition to Rüdiger Kilian comment and your links, I realized that the mistake is that I didn't know that EBSD is a variable not an MTEX function. I'm a beginner in Matlab and I wanna take the MTEX workshop this year.
Instead of using plotIPDF(meanO('copper').orientations,zvector), I used plotIPDF(EBSD('copper').orientations,zvector).
I did concatenate the variables but I was calling the last one only. Now the issue is resolved.

One thing I hope you can help me to understand. I attached two images for the IPF. I just wanna get the one with the color.key displayed. However, the code I copied (from misorientation axis documentations) has grain boundary in it. I don't feel it's write to used it.

gB_Cu = grains.boundary('copper','copper')

% computed the axes in specien coordinates
axes = gB_Cu.misorientation.axis

% define a color key
colorKey = HSVDirectionKey(axes);

% compute colors
color = colorKey.direction2color(axes);
**plot(colorKey); hold on;

plotIPDF(m('copper').orientations,zvector,'MarkerColor','black','MarkerSize',1)**




![untitled2](https://github.com/mtex-toolbox/mtex/assets/141929646/2fa2003d-937c-4b31-ab2e-90696006173c)
![untitled](https://github.com/mtex-toolbox/mtex/assets/141929646/04cd6bcf-cbee-4ae9-b77d-d8f1e8907eb3)

**Reply by zmichels** [reply]
I might be misunderstanding... but using the code/variables you've setup in that snippet... I *think* the plot you want can be made with:

plot(colorKey)

Then... save that figure instead/before adding the "hold on" and subsequent plotIPDF() line... which adds a layer of plotted points over top of the color key plot.
